### PR TITLE
Fix Github Action and modify supported xcode/macOS versions in both Github Action and Travis CI.

### LINKED ISSUE #566: "ci / test-mac (1.14, macos-10.15, 10.3)" and "ci / test-mac (1.13, macos-10.15, 10.3)" CI builds are failing
Error is

```
ld: unsupported tapi file type '!tapi-tbd' in YAML file '/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/lib/libSystem.tbd' for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Error: Process completed with exit code 2.
```

This has been discussed offline, filing to track.

I'll remove these builds from the required presubmit checks for now so that PRs are unblocked.

### PR DESCRIPTION
**Pull request by wyk9787**
Fixes #566 

* Issues to address:
  * Xcode version 10.3 has issue with macOS 10.15 in Github Action. See https://github.com/google/pprof/pull/560#issuecomment-699088881.
  * We do not currently have a comprehensive coverage on different xcode / macOS versions.

* Existing limitations
  * Github Action only supports macOS 10.15.
  * Github Action doesn't support Golang master branch.
  * Travis CI will time out if there are too many tests and there is no way to get around that.
  * Travis CI cannot freely switch xcode and macOS version. They are bundled together as `osx_image`, see https://docs.travis-ci.com/user/reference/osx/.

* Solution
  * Get rid of xcode 10.3 in Github Action.
  * Support {xcode 10.3, macOS 10.14.4} and {xcode 11.3, macOS 10.14.6} with Golang = {1.13, 1.14, master} in Travis since that will cover two minor xcode versions with macOS 10.14.
  * Support golang master for xcode 11.6 and xcode 11.2 since that is not covered by Github Action.

This PR also adds tests for xcode 12.0 and 11.7.

### CODE REVIEW

**Comment by wyk9787** [comment]
There are two expected checks that are not passing. It is because in the repo setting, we checked those two as required check before merging the PR. 

@kalyanac could you mark those two checks in the repo setting as not required and mark newly added xcode 12.0 related checks as required? 

I wish Github does a better job transitioning between adding and removing checks.

**Comment by wyk9787** [comment]
We should push out this PR soon so it can unblock the CI failures in other PRs.

**Review by nolanmar511 — APPROVED** [review]

**Comment by kalyanac** [comment]
Following are already not marked required

ci / test-mac (1.13, macos-10.15, 10.3) (pull_request)
ci / test-mac (1.14, macos-10.15, 10.3) (pull_request)

**Comment by kalyanac** [comment]
Please mention in the PR desc that this adds XCode 12.0 and 11.7 in GitHub Actions.

**Comment by wyk9787** [comment]
> Please mention in the PR desc that this adds XCode 12.0 and 11.7 in GitHub Actions.

Done.

**Comment by kalyanac** [comment]
GO 1.15 is released. We can drop 1.13. 
https://golang.org/doc/devel/release.html

**Review by kalyanac — APPROVED** [review]

**Pull request merged**
